toothpick-runtime/src/main/java/toothpick/configuration/Configuration.java

```diff
@@ -0,0 +1,105 @@
+package toothpick.configuration;
+
+import toothpick.Factory;
+import toothpick.MemberInjector;
+import toothpick.config.Binding;
+
+/**
+ * Strategy pattern that allows to change various behaviors of Toothpick.
+ * The default configuration is {@link #production()} and {@link #reflection()}.
+ * A custom configuration can be created and used by toothpick,
+ * it is even possible to use a composition of the built-in configurations.
+ */
+public class Configuration {
+
+  public static Configuration instance = new Configuration();
+
+  private ReflectionConfiguration reflectionConfiguration = new ReflectionOnConfiguration();
```

Comment:
It is a design decision we took. We can discuss it anyway.

If we use the reflectionfree configuration, some extra steps are needed (setting up registries, sending some annotation processing parameters).

So the default one is the one that will just run with no settings.